// dataOper/DAO/addressDAO.js
var Address = function () {
    this.ada = require('../adapter');
};


Address.prototype.getUserAddress = function (userPhone,callback) {
    var sql = "select * from t_address where userPhone = ? and del = '0' order by type";
    this.ada.all(sql,[userPhone],function (res,data) {
        if(res){
            callback(data);
        }
    })
};

Address.prototype.delAddress = function (userPhone,addressid,callback) {
    var sql = 'select * from t_address where addressId = ?';
    this.ada.all(sql,[addressid],function (res,data) {
        if(res){
            sql = "update t_address set del='1' where addressid = ?";
            this.ada.run(sql,[addressid],function (res) {
                if(res){
                    if(data[0].type == 1){
                        sql = "select * from t_address where userPhone = ? and `del` = '0'";
                        this.ada.all(sql,[userPhone],function (res,data) {
                            if(res){
                                if(data.length>0){
                                    var newid = data[0].addressId;
                                    this.changeDefault(userPhone,newid,function () {

                                    });
                                }

                            }
                        }.bind(this))
                    }
                    callback(true);
                }
                else {
                   callback(false);
                }
            }.bind(this))
        }

    }.bind(this))
};

Address.prototype.changeDefault =function (userPhone,addressid,callback) {
    var sql = "select * from t_address where userPhone = ? and type =1 and `del`='0'";
    this.ada.all(sql,[userPhone],function (res,data) {
        if(res){
            if(data.length>0){
                var oldid = data[0].addressId;
                sql = 'update t_address  set type = 2 where addressId = ?';
                this.ada.update(sql,[oldid],function () {});
            }
            sql = 'update t_address  set type = 1 where addressId = ?';
            this.ada.update(sql,[addressid],function (res) {
                if(res){
                    callback(true);
                }
                else{
                    callback(false);
                }
            });
        }
    }.bind(this))
};

Address.prototype.addAddress = function (userPhone,receiverName,receiverPhone,addressName,callback) {
    var sql = "select * from t_address where userPhone = ? and del='0'";
    var type=2;
    this.ada.all(sql,[userPhone],function (res,data) {
        if(res){
            if(data.length <= 0){
                type = 1;
            }
            sql = 'insert into t_address(userPhone,receiverName,receiverPhone,addressName,type) values(?,?,?,?,?)';
            this.ada.run(sql,[userPhone,receiverName,receiverPhone,addressName,type],function (res) {
                callback(res);
            })
        }
    }.bind(this));


};

Address.prototype.changeAddress = function (addressId,receiverName,receiverPhone,addressName,callback) {
    var sql = 'update t_address  set receiverName = ?,receiverPhone = ?,addressName = ? where addressId = ?';
    this.ada.run(sql,[receiverName,receiverPhone,addressName,addressId],function (res) {
        callback(res);
    })
};

module.exports = new Address();
