// dataOper/DAO/orderDAO.js
var Order = function () {
    this.ada = require('../adapter');
};

Order.prototype.getAll = function (callback) {
    var sql = 'select * from t_order';
    this.ada.all(sql,[],function (res,data) {
        if(res){
            callback(res,data);
        }
        else {
            callback(res);
        }
    })
};

Order.prototype.getUserOrder = function (userPhone,callback) {
    var sql =' select * from t_order a inner join t_trade b on a.tradeId = b.tradeId where a.userPhone = ? order by a.orderTime desc   ';

    this.ada.all(sql,[userPhone],function (res,data) {
        if(res){
            callback(data);
        }
    })
};

Order.prototype.getTradeOrder = function (tradeId, callback){
  var sql = 'select a.*,b.userName from t_order a inner join t_user b on a.userPhone = b.userPhone where tradeId = ? and orderStartId = 1';
  this.ada.all(sql,[tradeId],function (res,data) {
      if(res){
          callback(data);
      }
  })
};

Order.prototype.latelyOrder = function (callback) {
    var sql = 'select c.tradeName,b.userName,c.tradeId,a.orderTime from t_order a inner join t_user b on a.userPhone = b.userPhone INNER JOIN t_trade c on a.tradeId = c.tradeId where a.orderStartId = 1 order by a.orderTime desc limit 0,35';
    this.ada.all(sql,[],function (res,data) {
        if(res){
            callback(res,data);
        }
        else {
            callback(res);
        }
    })
};

Order.prototype.escOrder = function (orderId,callback) {
    var sql = 'update t_order set orderStartId = 2 where orderId = ?';
    this.ada.run(sql,[orderId],function (res) {
        callback(res);
    })
};

Order.prototype.getbuyNum = function (year,mon,day,userPhone,tradeId,callback) {
    var  sql = "select * from t_order where orderTime like ? and userPhone = ? and tradeId = ? and orderStartId != 0";
    var date = "%"+year+"-"+mon+"-"+day+"%";
    this.ada.all(sql,[date,userPhone,tradeId],function (res,data) {
        if(res){
            callback(res,data);
        }
        else {
            callback(res);
        }
    })
};

Order.prototype.fromredis = function (data,callback) {
    if(data!=null){
        var tradeId = data.tradeId;
        var userPhone = data.userPhone;
        var addressId = data.addressId;
        var orderStartId = data.orderStartId;
        var orderTime = data.orderTime;
        var orderMoney = data.orderMoney;
        var orderBuyNum = data.orderBuyNum;
        var orderAllMoney = data.orderAllMoney;

        var sql = "insert into t_order(tradeId,userPhone,addressId,orderStartId,orderTime,orderMoney,orderBuyNum,orderAllMoney) values(?,?,?,?,?,?,?,?)";
        this.ada.run(sql,[tradeId,userPhone,addressId,orderStartId,orderTime,orderMoney,orderBuyNum,orderAllMoney],function (res) {
            if(res){
                callback();
            }
        })
    }
};

Order.prototype.timeOut = function (orderId,callback) {
    var sql = "update t_order set orderStartId=0 where orderId = ? and orderStartId=2";
    this.ada.run(sql,[orderId],function () {
        if(callback != undefined){
            callback();
        }
    })
};

Order.prototype.checkOrder= function (orderId,callback) {
    var sql = "select * from t_order where orderId = ?";
    this.ada.all(sql,[orderId],function (res,data) {
        if(res){
            if(data[0].orderStartId == 2){
                callback(true,data[0].orderAllMoney);
            }
            else{
                callback(false);
            }
        }
    })
};

Order.prototype.checkOut = function (orderId,userPhone,orderMoney,callback) {
    var sql = 'update t_order set orderStartId=3 where orderId = ?';
    this.ada.run(sql,[orderId],function (res) {
        if(res){
            sql = 'select * from t_user where userPhone = ?';
            this.ada.all(sql,[userPhone],function (res,data) {
                if(res){
                    var userMoney = data[0].userMoney;
                    var newMoney = userMoney-orderMoney;
                    sql = 'update t_user set userMoney = ? where userPhone = ?';
                    this.ada.run(sql,[newMoney,userPhone],function (res) {
                        if(res){
                            callback();
                        }
                    })
                }
            }.bind(this))
        }
    }.bind(this))
};

module.exports = new Order();
