// dataOper/DAO/tradeDAO.js
var Trade = function () {
    this.ada = require('../adapter');
};

Trade.prototype.getAllandTime = function (callback) {
    var sql = 'select * from t_Time a inner join t_trade b on a.skId = b.tradeSKTimeId where b.state=0 and putawayTime < time(now()) and soldOutTime > time(now())';
    this.ada.all(sql,[],function (res,data) {
        if(res){
            callback(res,data);
        }
        else {
            callback(res);
        }
    })
};

Trade.prototype.getTradeByTime = function (timeID,callback) {
    var sql = 'select * from t_trade where tradeSKTimeId = ? and state=0 and putawayTime < time(now()) and soldOutTime > time(now())';
    this.ada.all(sql,[timeID],function (res,data) {
        if(res){
            callback(data);
        }
    })
};

Trade.prototype.getTradeByTimeC = function (timeID,tradeClassify,callback) {
    var sql = 'select * from t_trade where tradeSKTimeId = ? and tradeClassify=? and state=0 and putawayTime < time(now()) and soldOutTime > time(now())';
    this.ada.all(sql,[timeID,tradeClassify],function (res,data) {
        if(res){
            callback(data);
        }
    })
};

Trade.prototype.getTradeByID = function (tradeID,callback) {
    var sql = 'select * from t_trade where tradeID = ?';
    this.ada.all(sql,[tradeID],function (res,data) {
        if(res){
            callback(data);
        }
    })
};

Trade.prototype.fromredis = function (tradeInventory,tradeId) {
    var sql = "update t_trade set tradeInventory =? where tradeId = ?";
    this.ada.run(sql,[tradeInventory,tradeId],function () {
    })
};

module.exports = new Trade();
