// dataOper/DAO/userDAO.js
var User = function () {
    this.ada = require('../adapter');
};


User.prototype.getAll = function (callback) {
    var sql = 'select * from t_user';
    this.ada.all(sql,[],function (res,data) {
        if(res){
            callback(res,data);
        }
        else {
            callback(res);
        }
    })
};

User.prototype.checkUserPhone = function (UserPhone,callback) {
    var sql = 'select * from t_user where userPhone = ?';
    this.ada.all(sql,[UserPhone],function (res,data) {
        if(res){
            if(data.length < 1){
                callback(true);
            }
            else {
                callback(false);
            }
        }
    })
};

User.prototype.login = function (userPhone,psw,callback) {
    var sql = 'select * from t_user where userPhone = ? and userPsw = ?';
    this.ada.all(sql,[userPhone,psw],function (res,data) {
        if(res){
            if(data.length > 0){
                var userTop = {userPhone:data[0].userPhone,userName:data[0].userName,userMoney:data[0].userMoney,userImg:data[0].userImg,safetyCode:data[0].safetyCode};
                callback(true,userTop);
            }
            else {
                callback(false);
            }
        }
    })
};

User.prototype.sign = function (userPhone,psw,userName,safetyCode,callback) {
    var sql = 'insert into t_user(userPhone,userPsw,userName,userImg,userMoney,userRegisterTime,safetyCode)' +
                                        " values(?,?,?,'images\\headImg\\14.jpg',0,datetime('now','localtime'),?)";
    this.ada.run(sql,[userPhone,psw,userName,safetyCode],function (res) {
        callback(res);
    })
};

User.prototype.getUserInfo = function (userPhone, callback) {
    var sql = 'select * from t_user where userPhone = ?';

    this.ada.all(sql,[userPhone],function (res,data) {
        if(res){
            callback(data);
        }
    })
};

User.prototype.getMoney = function (userPhone,money,callback) {
    var sql = 'update t_user set userMoney = userMoney+? where userPhone = ?';
    this.ada.run(sql,[money,userPhone],function (res) {
        this.getUserInfo(userPhone,function (data) {
            var newmoney = data[0].userMoney;
            if(res){
                callback(true,newmoney);
            }
            else {
                callback(false,newmoney);
            }
        });

    }.bind(this))
};

User.prototype.changeNickname = function (userPhone,nickname,callback) {
    var sql = 'update t_user set userName = ? where userPhone = ?';
    this.ada.run(sql,[nickname,userPhone],function (res) {
        if (res) {
            callback(true);
        }
        else {
            callback(false);
        }
    })
};

User.prototype.changePassword = function (userPhone,psw,callback) {
    var sql = 'update t_user set userPsw = ? where userPhone = ?';
    this.ada.run(sql,[psw,userPhone],function (res) {
        if (res) {
            callback(true);
        }
        else {
            callback(false);
        }
    })
};

User.prototype.checkUser = function (userPhone,safetyCode,callback) {
    var sql = 'select * from t_user where userPhone = ? and safetyCode = ?';
    this.ada.all(sql,[userPhone,safetyCode],function (res,data) {
        if(res){
            if(data.length > 0){
                callback(true);
            }
            else {
                callback(false);
            }
        }
    })
};

User.prototype.checkUserMoney = function (userPhone,Money,callback) {
    var sql = 'select * from t_user where userPhone = ?';
    this.ada.all(sql,[userPhone],function (res,data) {
        if(res){
            if(data[0].userMoney >= Money){
                callback(true);
            }
            else {
                callback(false);
            }
        }
    })
};


module.exports = new User();
